pkg/io/opener.go

```diff
@@ -94,29 +131,74 @@ func (o opener) openGCS(path string) (*storage.ObjectHandle, error) {
 	if p.Object() == "" {
 		return nil, errors.New("object name is empty")
 	}
-	return o.gcs.Bucket(p.Bucket()).Object(p.Object()), nil
+	return o.gcsClient.Bucket(p.Bucket()).Object(p.Object()), nil
+}
+
+// getBucket opens a bucket
+// The storageProvider is discovered based on the given path.
+// The buckets are cached per bucket name. So we don't open a bucket multiple times in the same process
+func (o *opener) getBucket(ctx context.Context, path string) (*blob.Bucket, string, error) {
+	_, bucketName, relativePath, err := providers.ParseStoragePath(path)
+	if err != nil {
+		return nil, "", fmt.Errorf("could not get bucket: %w", err)
+	}
+
+	o.cachedBucketsMutex.Lock()
+	defer o.cachedBucketsMutex.Unlock()
+	if bucket, ok := o.cachedBuckets[bucketName]; ok {
+		return bucket, relativePath, nil
+	}
+
+	bucket, err := providers.GetBucket(ctx, o.s3Credentials, path)
+	if err != nil {
+		return nil, "", err
+	}
+	o.cachedBuckets[bucketName] = bucket
+	return bucket, relativePath, nil
 }
 
 // Reader will open the path for reading, returning an IsNotExist() error when missing
-func (o opener) Reader(ctx context.Context, path string) (io.ReadCloser, error) {
-	g, err := o.openGCS(path)
+func (o *opener) Reader(ctx context.Context, path string) (io.ReadCloser, error) {
+	if strings.HasPrefix(path, "gcs://") || strings.HasPrefix(path, "gs://") {
```

Comment:
Please drop support for `gcs://`, the only commonly recognized scheme is `gs://`

```console
$ gsutil ls gcs://
InvalidUrlError: Unrecognized scheme "gcs".
```

Reply:
Done